crates/router/src/connector/checkout.rs

```diff
@@ -1185,12 +1213,22 @@ impl api::IncomingWebhook for Checkout {
         &self,
         request: &api::IncomingWebhookRequestDetails<'_>,
     ) -> CustomResult<serde_json::Value, errors::ConnectorError> {
-        let details: checkout::CheckoutWebhookObjectResource = request
+        let event_type_data: checkout::CheckoutWebhookEventTypeBody = request
             .body
-            .parse_struct("CheckoutWebhookObjectResource")
-            .change_context(errors::ConnectorError::WebhookResourceObjectNotFound)?;
-
-        Ok(details.data)
+            .parse_struct("CheckoutWebhookBody")
+            .change_context(errors::ConnectorError::WebhookEventTypeNotFound)?;
+        let resource_object = if checkout::is_chargeback_event(&event_type_data.transaction_type) {
```

Comment:
Only in case of payments event it should be payment response. In other cases we should return the object as it is